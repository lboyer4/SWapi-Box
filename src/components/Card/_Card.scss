article {
	height: 200px;
	width: 150px;
	border: 2px solid blue;
	background-color: #fff;
	color: black;
}