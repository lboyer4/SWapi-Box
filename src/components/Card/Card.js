import React, { Component } from 'react';
import './_Card.scss';


class Card extends Component {
	constructor() {
		super();
		this.state = {
			favorited: false
		}
	}

	handleClick = (e) => {
		const favorited = !this.state.favorited
		this.setState({favorited})
		if (favorited) {
			e.target.parentElement.classList += 'favorited';
		} else {
			e.target.parentElement.classList = '';
		}	
	}

	render() {
		const cardData = Object.keys(this.props).map(key => {
			return <h5>{ this.props[key] }</h5>
		})
		return (
			<article>
				{cardData}
				<button onClick={this.handleClick}>favorite</button>		
			</article>
		)
	}
}


export default Card;