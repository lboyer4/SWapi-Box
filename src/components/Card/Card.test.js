import React from 'react';
import { shallow } from 'enzyme';
import { Card } from './Card.js';

describe('Card', () => {
	let wrapper;

	it('should match the snapshot', () => {
		const wrapper = shallow(<Card />);
		expect(wrapper).toMatchSnapshot();
	});

	it('updates the state of the favorited', () => {
		const mockEvent = jest.fn()
		wrapper.setState({favorited: false})
		wrapper.instance().handleClick(mockEvent);
		expect(wrapper.state().favorited).toEqual(true)
	});

	it('should call add favorite', () => {
		const mockEvent = jest.fn()
		wrapper.setState({favorited: true})
		wrapper.instance().handleClick(mockEvent);
		expect(addFavorite).toHaveBeenCalled()
	});

	it('should call remove favorite', () => {
		const mockEvent = jest.fn()
		wrapper.setState({favorited: false})
		wrapper.instance().handleClick(mockEvent);
		expect(addFavorite).toHaveBeenCalled()
	});

})