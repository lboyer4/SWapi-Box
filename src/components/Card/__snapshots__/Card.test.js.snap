// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`Card should should match snapshot if planet is passed in 1`] = `ShallowWrapper {}`;

exports[`Card should should match snapshot if vehicle is passed in 1`] = `ShallowWrapper {}`;
